### DISCUSSION TITLE
%EC%84%B1%EB%B9%88-%ED%81%B4%EB%9F%BD-%EB%AF%B8%EC%85%986/

### DISCUSSION THREAD

**Discussion by giscus** [SungbinYang/SungbinYang.github.io#10] [Comments]
# %EC%84%B1%EB%B9%88-%ED%81%B4%EB%9F%BD-%EB%AF%B8%EC%85%986/

양성빈의 블로그

https://sungbin.kr/%EC%84%B1%EB%B9%88-%ED%81%B4%EB%9F%BD-%EB%AF%B8%EC%85%986/

**Comment by ljh468** [comment]
LGTM 👍
스코프는 변수가 접근할 수 있는 영역을 의미하고, 생명 주기는 변수가 메모리에 존재하는 기간을 의미한다.!! `완전 깔끔한 정리`네요.
고급 문제 3번은 같은 의미이지만, num1과num2의 합이 float을 초과할 수 있기 때문에 float으로 형변환을 하면 데이터 손실이 발생한다. 라고 적으면 더 자연스러운 문장이 될 것 같아요. 고생하셨습니다.!! 굳굳

**Reply by SungbinYang** [reply]
오호 고급3번은 말씀하신데로 변경해보는 것도 좋을 것 같군요! 피드백 감사드립니다 :)

**Comment by moonnahye** [comment]
오 정리잘해주셔서 이해하기 좋아요!! 그런데 자바가 낮은 정밀도 타입에서 높은 정밀도 타입으로 바뀐다는게 어떤의미인가요..?! 실행되는 시점인가..?!

**Reply by SungbinYang** [reply]
쉽게 생각해서 낮은 정밀도 타입에서 높은 정밀도 타입으로 바뀐다는 것은 자동형변환에 대한 예시를 생각하시면 좋을 것 같아요!

``` java
long value = 12;
```

위와 같다고 볼 수 있습니다! 실행되는 시점이 아닌 컴파일 시점에 변경되는거구요! 이해가 되셨을까요?

**Comment by tdronnie** [comment]
스코프랑 생명주기를 함께 생각해볼 수 있어서 좋았습니다 깔끔한 정리 잘봤습니다!!

**Reply by SungbinYang** [reply]
감사합니다 ㅎㅎ
